main.go

```diff
@@ -0,0 +1,82 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"cloud.google.com/go/pubsub"
+)
+
+func main() {
+
+	err := publishMessage("pubsub1", "testing1", data{
+		messageID:    "1",
+		requestBody:  "api request body",
+		responseBody: "api response body",
+	})
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+}
+
+type data struct {
+	messageID		string
+	requestBody		string
+	responseBody	string
+}
+
+// PublishMessage publishes messages to already created topics
+func publishMessage(projectID, topicID string, msg data) error {
+	ctx, cancel := context.WithTimeout(context.Background(), 30 * time.Second)
+	defer cancel()
+
+	os.Setenv("PUBSUB_EMULATOR_HOST", "localhost:8085")
+
+	client, err := pubsub.NewClient(ctx, projectID)
+	if err != nil {
+		fmt.Println(err)
+		return err
+	}
+	defer client.Close()
+
+	topic := client.Topic(topicID)
+	defer topic.Stop()
```

Comment:
When you create the function that creates this topic, it should return a callback function which the user can call to trigger traffic.Stop()

Why all this is important, is that we will be publishing messages hundreds of thousands of times per minute. And we don't want to be initializing a topic for each message. We want to initialize the topic once, and then reuse it every time we need to publish the message. Then when everything is completely done, the caller can shutdown the topic